metadata-db-app/src/cmr/metadata_db/data/memory_db.clj

```diff
@@ -131,6 +132,111 @@
        :existing-concept-id existing-concept-id
        :existing-native-id existing-native-id})))
 
+(defn- concept-id-not-in-list?
+  "Check if the concept-id is not in the list of concept-ids."
+  [list concept-id]
+  (not (some #(= concept-id %) list)))
+
+(defn- concept-id-in-list?
+  "Check if the concept-id is in the list of concept-ids."
+  [list concept-id]
+  (some #(= concept-id %) list))
+
+(defn validate-collection-not-associated-with-another-variable-with-same-name
+  "Validates that collection in the concept is not associated with a different
+  variable, which has the same name as the variable in the concept.
+  Returns nil if valid and an error response if invalid."
+  [db concept]
+  (let [variable-concept-id (get-in concept [:extra-fields :variable-concept-id])
+        associated-concept-id (get-in concept [:extra-fields :associated-concept-id])]
+    (when (and variable-concept-id associated-concept-id)
+      (let [;;get variable name for the variable-concept-id from cmr_variables.
+            variable (->> @(:concepts-atom db)
+                            (filter #(= variable-concept-id (:concept-id %)))
+                            first)
```

Comment:
Is it possible to have multiples of the same concept id in the collection?

Reply:
Yeah. at this point, it gets all the rows for all the revisions.